File: 6. 심화 1/1157.py
string = input().upper()
result = [0 for i in range(26)]
Max = 0

if (1 <= len(string) <= 1000000):
    for i in list(string):
        result[ord(i)-66] += 1
    for i in result:
        Max = Max if Max > i else i
    if result.count(Max) >= 2:
        print('?')
    else:
        print(chr(result.index(Max)+66))

File: 6. 심화 1/1157.c
#include <stdio.h>

int main() {
    char str[1000000], result = '?';
    int size, max = 0, alpha[26] = {0};
    scanf("%s", str);

    for (size = 0; str[size] != '\0'; size++) {
        if (str[size] <= 'Z') {
            alpha[str[size]-'A'] += 1;
        } else {
            alpha[str[size]-'a'] += 1;
        }
    }

    for (int i = 0; i < 26; i++) {
        if (alpha[i] > max) {
            max = alpha[i];
            result = i + 'A';
        } else if (alpha[i] == max) {
            result = '?';
        }
    }

    printf("%c", result);

    return 0;
}

